### PR TITLE
NIFIREG-47 Improvements to nifi-registry-client

### PR DESCRIPTION
**Pull request by bbende**
- Using separate Jersey version for client only
- Added support for proxied entities
- Fixed deserializing of BucketItems
- Cleaned up methods with unnecessary parameters

### CODE REVIEW

**Comment by bbende** [comment]
Not sure why one travis job failed, all tests passing for me locally

File: nifi-registry-client/src/main/java/org/apache/nifi/registry/client/impl/JerseyBucketClient.java [hunk 1]
```diff
@@ -140,7 +150,8 @@ public Fields getFields() throws NiFiRegistryException, IOException {
                 target = target.queryParam("sort", sortParam.toString());
             }
 
-            return Arrays.asList(target.request().get(Bucket[].class));
+            return getRequestBuilder(target).get(List.class);
```
Comment (markap14) [inline nifi-registry-client/src/main/java/org/apache/nifi/registry/client/impl/JerseyBucketClient.java]: I think this should be done as above in the other getAll() call no? Getting a Bucket[] and then returning emptyList or Arrays.asList?

**Comment by markap14** [comment]
@bbende code review all looks good except for the one inline comment I made. Thanks!

**Comment by bbende** [comment]
@markap14 thanks for reviewing! pushed up a commit to address your comment, let me know if you are good and want me to merge

**Comment by bbende** [comment]
This was merged by Mark, closing the PR

**Pull request closed without merging**